content/v2.0/introduction/getting-started.md

```diff
@@ -138,29 +162,1261 @@ Visit your application in the browser at [http://localhost:2300](http://localhos
+
+In the file `spec/requests/root_spec.rb`, Hanami provides a request spec for the "Hello from Hanami" message we've seen in the browser.
+
+```ruby
+# spec/requests/root_spec.rb
+
+# frozen_string_literal: true
+
+RSpec.describe "Root", type: :request do
+  it "is successful" do
+    get "/"
+
+    # Find me in `config/routes.rb`
+    expect(last_response).to be_successful
+    expect(last_response.body).to eq("Hello from Hanami")
+  end
+end
+```
+
+Let's run that spec now to prove that it works:
+
+```shell
+bundle exec rspec spec/requests/root_spec.rb
+```
+
+You should see:
+
+```shell
+Randomized with seed 37065
+
+Root
+  is successful
+
+Finished in 0.01986 seconds (files took 0.70103 seconds to load)
+1 example, 0 failures
+
+Randomized with seed 37065
+```
+
+Let's change the "Hello from Hanami" message to "Welcome to Bookshelf". First, we'll adjust our spec:
+
+```ruby
+# spec/requests/root_spec.rb
+
+# frozen_string_literal: true
+
+RSpec.describe "Root", type: :request do
+  it "is successful" do
+    get "/"
+
+    # Find me in `config/routes.rb`
+    expect(last_response).to be_successful
+    expect(last_response.body).to eq("Welcome to Bookshelf")
+  end
+end
+```
+
+As we expect, when we run the spec again, it fails:
+
+```shell
+Randomized with seed 65034
+
+Root
+  is successful (FAILED - 1)
+
+Failures:
+
+  1) Root is successful
+     Failure/Error: expect(last_response.body).to eq("Welcome to Bookshelf")
+
+       expected: "Welcome to Bookshelf"
+            got: "Hello from Hanami"
+
+       (compared using ==)
+     # ./spec/requests/root_spec.rb:9:in `block (2 levels) in <top (required)>'
+
+Finished in 0.04572 seconds (files took 0.72148 seconds to load)
+1 example, 1 failure
+
+Failed examples:
+
+rspec ./spec/requests/root_spec.rb:4 # Root is successful
+
+Randomized with seed 65034
+```
+
+To fix this, let's open our application's routes file at `config/routes.rb`:
+
+```ruby
+# config/routes.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Routes < Hanami::Routes
+    root { "Hello from Hanami" }
+  end
+end
+```
+
+This `Bookshelf::Routes` class contains the configuration for our application's router. Routes in Hanami are comprised of a HTTP method, a path, and an endpoint to be invoked, which is usually a Hanami action. (See the [Routing guide](/v2.0/routing/overview/) for more information).
+
+We'll take a look at adding more routes in a moment, but for now let's get our spec to pass. The above `Bookshelf::Routes` class contains only one route, the `root` route, which handles `GET` requests for `"/"`.
+
+Rather than invoking an action, this route is configured to invoke a block, which returns "Hello from Hanami".
+
+Blocks are convenient, but let's adjust our root route to invoke an action instead:
+
+```ruby
+# config/routes.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Routes < Hanami::Routes
+    root to: "home.index"
```

Comment:
Should this actually be `"home.show"`? The home page is a singleton resource and we're showing it, rather than providing a list of home pages.

Reply:
Oh yep, adjusted to "home.show", thanks.